crates/sp-subspace-mmr/src/runtime_interface.rs

```diff
@@ -18,17 +18,36 @@ use subspace_core_primitives::BlockNumber;
 #[runtime_interface]
 pub trait SubspaceMmrRuntimeInterface {
     /// Returns the MMR leaf.
+    #[cfg_attr(feature = "runtime-benchmarks", allow(unused_variables))]
     fn get_mmr_leaf_data(&mut self, consensus_block_hash: H256) -> Option<LeafData> {
-        self.extension::<SubspaceMmrExtension>()
-            .expect("No `SubspaceMmrExtension` associated for the current context!")
-            .get_mmr_leaf_data(consensus_block_hash)
+        #[cfg(not(feature = "runtime-benchmarks"))]
+        {
+            self.extension::<SubspaceMmrExtension>()
+                .expect("No `SubspaceMmrExtension` associated for the current context!")
+                .get_mmr_leaf_data(consensus_block_hash)
+        }
+
+        // TODO: check this implementation has a greater or similar cost to the real implementation
```

Comment:
Please avoid TODOs that have no clear next path. 

if you are unsure, then you can play around and find the if these random values have a similar or different cost

Reply:
I don't know how to check the cost for calls into the `SubspaceMmrExtension`.

How would you do it?

Reply:
I added some more stub code, but we're blocked from using `SubspaceMmrExtension` in benchmarks by https://github.com/paritytech/polkadot-sdk/issues/137

Since this change is a lower priority than staking, I suggest we merge what we have as an improvement, and come back to it later.

As an alternative, we could use the upstream weights instead?
https://github.com/paritytech/polkadot-sdk/blob/6824a6782354e274cf308be314f142551c5457ab/polkadot/runtime/rococo/src/weights/pallet_mmr.rs
They have significantly higher CPU, proof, and read weights, but slightly lower writes.